contracts/governance/ProposalValidator.sol

```diff
@@ -185,16 +191,69 @@ contract ProposalValidator is IProposalValidator {
     view
     returns (bool)
   {
-    IKyberGovernance.ProposalWithoutVote memory proposal = governance.getProposalById(
-      proposalId
-    );
-    uint256 votingSupply = IGovernanceStrategy(proposal.strategy).getTotalVotingSupplyAt(
-      proposal.startBlock
-    );
-
-    return (proposal.forVotes.mul(ONE_HUNDRED_WITH_PRECISION).div(votingSupply) >
-      proposal.againstVotes.mul(ONE_HUNDRED_WITH_PRECISION).div(votingSupply).add(
+    IKyberGovernance.ProposalWithoutVote memory proposal = governance.getProposalById(proposalId);
+    if (proposal.proposalType != ProposalType.Binary) return false;
+    return (proposal.voteCounts[1].mul(ONE_HUNDRED_WITH_PRECISION).div(proposal.maxVotingPower) >
+      proposal.voteCounts[0].mul(ONE_HUNDRED_WITH_PRECISION).div(proposal.maxVotingPower).add(
         VOTE_DIFFERENTIAL
       ));
   }
+
+  function getWinningOption(IKyberGovernance governance, uint256 proposalId)
+    external
+    view
+    override
+    returns (uint256 winningOption)
+  {
+    IKyberGovernance.ProposalWithoutVote memory proposal = governance.getProposalById(proposalId);
+    
+    uint256 winningVoteCount;
+    (winningOption, winningVoteCount) = getWinningOptionData(proposal.voteCounts);
```

Comment:
should handle differently for binary proposal? since it is using a different mechanism to check for proposal is passed or not

Reply:
hmmm, binary proposals should use the `isProposalPassed` function to check. This function is for generic proposals. I can either
1) return 0 for binary proposals
2) call isProposalPassed and return the correct value

Reply:
yeah we need to be clear about returning value for each proposal type, may be change it to `getGenericProposalWinningOption` and `isBinaryProposalPassed`